renku/core/template/template.py

```diff
@@ -420,6 +420,10 @@ def fetch(cls, source: Optional[str], reference: Optional[str]) -> "RepositoryTe
         try:
             repository = clone_repository(url=source, path=path, checkout_revision=reference, install_lfs=False)
         except errors.GitError as e:
+            if "Cannot checkout reference" in str(e):
+                raise errors.TemplateMissingReferenceError(
+                    f"Cannot find the reference {reference} in the template repository from {source}"
```

Comment:
`reference {reference} -> reference '{reference}'` so that it's more visible what the reference is.